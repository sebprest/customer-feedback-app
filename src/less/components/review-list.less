.review {
    margin-bottom: @margin-lg;
}
