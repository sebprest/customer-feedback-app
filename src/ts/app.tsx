import React from "react";

const App: React.FunctionComponent = React.memo(() => <h1>Hello</h1>);

export default App;
