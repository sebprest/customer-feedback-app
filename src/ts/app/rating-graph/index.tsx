import { Line, LineChart, ResponsiveContainer, YAxis } from "recharts";
import React from "react";
import { connect } from "react-redux";

import { ReviewsState } from "^/common/types";
import { StoreState } from "^/store/types";

interface StateProps {
  reviews: ReviewsState;
}

type Props = StateProps;

export const RatingGraph: React.FunctionComponent<Props> = React.memo(
  (props) => {
    return (
      <div className="rating-graph">
        <ResponsiveContainer>
          <LineChart data={props.reviews}>
            <YAxis />
            <Line type="monotone" dataKey={"rating"} animationDuration={0} />
          </LineChart>
        </ResponsiveContainer>
      </div>
    );
  }
);

const mapStateToProps = ({ reviews }: StoreState): StateProps => {
  return {
    reviews,
  };
};

export default connect(mapStateToProps)(RatingGraph);
