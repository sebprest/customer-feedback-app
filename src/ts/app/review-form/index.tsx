import React, { useState } from "react";
import { connect } from "react-redux";

import { addReview } from "./actions";
import ReviewFormFields from "./review-form-fields";

type Props = {
  addReview: typeof addReview;
};

export const ReviewForm: React.FunctionComponent<Props> = React.memo(
  (props) => {
    const defaultValues = { name: "", email: "", rating: 0, comment: "" };
    const [values, setValues] = useState(defaultValues);

    const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
      const targetName = event.target.name;
      const targetValue = event.target.value;

      setValues((values) => ({ ...values, [targetName]: targetValue }));
    };

    const onAddReview = (event: React.FormEvent<HTMLFormElement>) => {
      event.preventDefault();
      props.addReview(values);
      setValues(() => defaultValues);
    };

    return (
      <ReviewFormFields
        addReview={onAddReview}
        updateFormState={handleInputChange}
        formState={values}
      />
    );
  }
);

export default connect(null, { addReview })(ReviewForm);
