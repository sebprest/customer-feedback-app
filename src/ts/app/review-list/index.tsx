import React from "react";
import { connect } from "react-redux";

import { ReviewsState } from "^/common/types";
import { StoreState } from "^/store/types";
import Review from "./review";

interface StateProps {
  reviews: ReviewsState;
}

type Props = StateProps;

export const ReviewList: React.FunctionComponent<Props> = React.memo(
  (props) => (
    <div className="review-list">
      <h2>All reviews ({props.reviews.length}):</h2>
      {props.reviews.map((review, index) => (
        <Review review={review} key={index} />
      ))}
    </div>
  )
);

const mapStateToProps = ({ reviews }: StoreState): StateProps => {
  return {
    reviews,
  };
};

export default connect(mapStateToProps)(ReviewList);
