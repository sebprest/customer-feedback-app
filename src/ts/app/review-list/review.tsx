import React from "react";

import { Review as TReview } from "^/common/types";

type Props = {
  review: TReview;
};

export const Review: React.FunctionComponent<Props> = React.memo((props) => (
  <div className="review">
    {props.review.name} ({props.review.email}) - Rating: {props.review.rating}:
    <div className="review-comment">{props.review.comment}</div>
  </div>
));

export default Review;
