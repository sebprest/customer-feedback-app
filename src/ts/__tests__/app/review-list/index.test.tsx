jest.mock("^/app/review-list/review", () => (props: unknown) => (
  <div {...props}>Review</div>
));

import React from "react";
import renderer from "react-test-renderer";

import { FAKE_REVIEW } from "^/__tests__/__helpers__/fake-review";
import { ReviewList } from "^/app/review-list";

describe("ReviewList", () => {
  it("should render a review component for each review provided", () => {
    const reviewForm = renderer.create(<ReviewList reviews={[FAKE_REVIEW]} />);

    expect(reviewForm).toMatchSnapshot();
  });
});
