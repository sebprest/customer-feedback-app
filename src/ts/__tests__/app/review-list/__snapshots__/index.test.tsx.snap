// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewList should render a review component for each review provided 1`] = `
<div
  className="review-list"
>
  <h2>
    All reviews (
    1
    ):
  </h2>
  <div
    review={
      Object {
        "comment": "fake comment",
        "email": "fake email",
        "name": "fake name",
        "rating": 5,
      }
    }
  >
    Review
  </div>
</div>
`;
